### ISSUE TITLE
Some test failures not detected by Teamcity (no output from NUnit)

### ISSUE THREAD

**Issue by oskarb** [nunit/teamcity-event-listener#70]
Nunit console 3.10. Some test failures are reported in the output from each test as each test finishes but failures from other test cases are only reported in the summary at the end after all tests have finished. Parallel execution not used. Failures that are only reported in the summary causes Teamcity to not detect these failures.

The failures that are reported for each affected test case actually occurs in OneTimeSetup. The failures (for other test cases) that are only reported in the summary occur in regular [Setup], or in the test case itself.

What can cause this?

I have create a bug for Teamcity here, where I provided sample log output to show the problem:
https://youtrack.jetbrains.com/issue/TW-61442

**Comment by oskarb** [comment]
@NikolayPianikov Further experiments suggest that failures are not reported in Teamcity for tests where the standard output from the test doesn't end with a new line. In this case, the testFailed TC service message is tacked on to the end of the last line of test output and it seems this might make Teamcity miss them - though when looking at the build log in TC the service message is not visible there. So TC detects the service message to avoid showing it in the build log, but it fails to "act" upon the contents of the message?

TC docs state that:
> To be processed by TeamCity, they need to be written to the standard output stream of the build, i.e. printed or echoed from a build step. It is recommended to output a single service message per line (i.e. divide messages with newline symbol(s))

So the docs does *not* really say that service message must be on a separate line. Only that it's recommended. In either case, I'm not outputting service messages - that is handled by this extension. It's brittle if I have to remember to design the tests to always end their output with a newline to avoid TC failing to detect the failures.

**Comment by NikolayPianikov** [comment]
@oskarb Could you share some simple solution to reproduce this issue, also please share actual and expected results for this solution.

**Comment by oskarb** [comment]
@NikolayPianikov 

The expected result is that TC will report 6 failed tests for the following code snippet. The actual result is that only TestsWithErrorInOneTimeSetup_WithStdOut.Test1 is reported as failed in TC GUI. The buildlog will say in the NUnit summary that all of them failed. Tests are run using the NUnit TC build step.

If the four Console.Write are commented out, TC will correctly report 6 failures.

Notice that when the error occurs in OneTimeSetup, the error will be duplicated and reported separately for Test1 and Test2 in TC, if Console.Write call is commented out. This seems expected and good. If the write to console is present, TC will detect the failure for Test1 but not for Test2. I'm guessing the standard output from OneTimeSetup only occurs once, even if the exception is reported twice, and therefore it messes up the service message for one of the cases but not the other one.



```
using System;
using NUnit.Framework;

namespace FailuresNotDetectedByTeamcity
{
    [TestFixture]
    public class TestsWithErrorInOneTimeSetup_WithStdOut
    {
        private static int _execCount;

        [OneTimeSetUp]
        public void OneTimeSetup()
        {
            _execCount++;
            Console.Write("TEST OUTPUT WITH NO NEWLINE");
            throw new Exception($"Error in OneTimeSetup. Exec count: {_execCount}");
        }

        [Test]
        public void Test1()
        {
        }

        [Test]
        public void Test2()
        {
        }
    }


    [TestFixture]
    public class TestsWithErrorInSetup_WithStdOut
    {
        private static int _execCount;

        [SetUp]
        public void PerTestSetup()
        {
            _execCount++;
            Console.Write("TEST OUTPUT WITH NO NEWLINE");
            throw new Exception($"Error in PerTestSetup. Exec count: {_execCount}");
        }

        [Test]
        public void Test1()
        {
        }

        [Test]
        public void Test2()
        {
        }
    }


    [TestFixture]
    public class TestsWithErrorInTest_WithStdOut
    {
        private static int _execCount;

        [Test]
        public void Test1()
        {
            _execCount++;
            Console.Write("TEST OUTPUT WITH NO NEWLINE");
            throw new Exception($"Error in Test1. Exec count: {_execCount}");
        }

        [Test]
        public void Test2()
        {
            _execCount++;
            Console.Write("TEST OUTPUT WITH NO NEWLINE");
            throw new Exception($"Error in Test2. Exec count: {_execCount}");
        }
    }
}
```

**Comment by NikolayPianikov** [comment]
@oskarb Now the issue is reproducing. Thank you very much

**Comment by NikolayPianikov** [comment]
@oskarb If you have TeamCity 2019 installed try updating [plugin](https://drive.google.com/open?id=1USxeI6mwv4ji0-Sy3B8crYYuCbGLvFHs)

**Comment by oskarb** [comment]
@NikolayPianikov I've tried the patched plugin and both the test cases I posted above to reproduce and the actual currently failing test cases now seem to be reported correctly. Thanks!

**Comment by schuettecarsten** [comment]
Do you release the plugin update for the public?

**Comment by NikolayPianikov** [comment]
@schuettecarsten it will be available in the next patch release 2019.1.x and in the EAP 2019.2. Also you could install from the link above, it was built as a part of 2019.1.x and passed all tests
